Fix parallel build issue

The config.h header file must be created before being used to build
source files that include it, especially version.c. In order for this
to happen even in highly-parallel builds, we add a dependency of all
object files on config.h, in order to ensure it gets generated before
make attempts to build the object files.

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>

Index: b/Makefile
===================================================================
--- a/Makefile
+++ b/Makefile
@@ -37,6 +37,8 @@
 
 all: config.h $(PPTP_BIN) pptpsetup.8
 
+$(PPTP_OBJS): config.h
+
 $(PPTP_BIN): $(PPTP_OBJS) $(PPTP_DEPS)
 	$(CC) -o $(PPTP_BIN) $(PPTP_OBJS) $(LDFLAGS) $(LIBS)
 
